src/lib/update-lock.ts

```diff
@@ -119,74 +108,65 @@ export async function lockAll<T extends unknown>(
- *
  * TODO: Remove skipLock as it's not a good interface. If lock is called it should try to take the lock
  * without an option to skip.
  */
-export function lock<T extends unknown>(
+export async function lock<T extends unknown>(
 	appId: number,
 	{ force = false, skipLock = false }: { force: boolean; skipLock?: boolean },
 	fn: () => Resolvable<T>,
-): Bluebird<T> {
+): Promise<T> {
 	if (skipLock || appId == null) {
-		return Bluebird.resolve(fn());
+		return Promise.resolve(fn());
 	}
 
-	const takeTheLock = () => {
-		return config
-			.get('lockOverride')
-			.then((lockOverride) => {
-				return writeLock(appId)
-					.tap((release: () => void) => {
-						const lockDir = getPathOnHost(lockPath(appId));
-						return Bluebird.resolve(fs.readdir(lockDir))
-							.catchReturn(ENOENT, [])
-							.mapSeries((serviceName) => {
-								return Bluebird.mapSeries(
-									lockFilesOnHost(appId, serviceName),
-									(tmpLockName) => {
-										return (
-											Bluebird.try(() => {
-												if (force || lockOverride) {
-													return lockfile.unlock(tmpLockName);
-												}
-											})
-												.then(() => {
-													return lockfile.lock(tmpLockName, LOCKFILE_UID);
-												})
-												// If lockfile exists, throw a user-friendly error.
-												// Otherwise throw the error as-is.
-												// This will interrupt the call to Bluebird.using, so
-												// dispose needs to be called even though it's referenced
-												// by .disposer later.
-												.catch((error) => {
-													return dispose(release, appId).throw(
-														lockfile.LockfileExistsError.is(error)
-															? new UpdatesLockedError(
-																	`Lockfile exists for ${JSON.stringify({
-																		serviceName,
-																		appId,
-																	})}`,
-															  )
-															: (error as Error),
-													);
-												})
-										);
-									},
-								);
-							});
-					})
-					.disposer((release: () => void) => dispose(release, appId));
-			})
-			.catch((err) => {
-				throw new InternalInconsistencyError(
-					`Error getting lockOverride config value: ${err?.message ?? err}`,
-				);
-			});
-	};
+	const lockDir = getPathOnHost(lockPath(appId));
+	let lockOverride: boolean;
+	try {
+		lockOverride = await config.get('lockOverride');
+	} catch (err) {
+		throw new InternalInconsistencyError(
+			`Error getting lockOverride config value: ${err?.message ?? err}`,
+		);
+	}
 
-	const disposer = takeTheLock();
+	// Acquire write lock for appId
+	await writeLock(appId);
 
-	return Bluebird.using(disposer, fn as () => PromiseLike<T>);
+	// Get list of service folders in lock directory
+	let serviceFolders: string[] = [];
+	try {
+		serviceFolders = await fs.readdir(lockDir);
+	} catch (e) {
+		log.error(`Error getting lock directory contents - ${e}`);
+	}
+
+	for await (const service of serviceFolders) {
```

Comment:
don't we have to wait for the lock to complete since it's async ? The reason I am doing this refactor is because I cannot reliable write tests when promises are not awaited.

Reply:
I meant if `await` is used in the next line of both `for await`s, are the awaits here necessary or are they redundant?

Reply:
I see. I never used this format, just added it because the linter yelled at me but you're right that only 1 await is needed!